src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -142,131 +147,239 @@ class CrossDeviceLinkUI extends Component {
     }
   }
 
   handleSMSError = (error) => {
     this.clearSendLinkClickTimeout()
-    this.setState({sending: false})
+    this.setState({ sending: false })
     this.props.triggerOnError(error)
     status === 429 ? this.setError('SMS_OVERUSE') : this.setError('SMS_FAILED')
   }
 
-  handleSendLinkClick = () => {
+  handleSendSmsLinkClick = () => {
     if (!this.props.sms.valid) {
       this.clearSendLinkClickTimeout()
-      this.setState({validNumber: false})
+      this.setState({ validNumber: false })
     } else if (!this.sendLinkClickTimeoutId) {
-      this.sendLinkClickTimeoutId = setTimeout(this.sendSms, 500);
+      this.sendLinkClickTimeoutId = setTimeout(this.sendSms, 500)
     }
   }
 
   sendSms = () => {
-    this.setState({sending: true})
+    this.setState({ sending: true })
     // add a quick note that this will send a production SMS, so non-production
     // environment users will need to amend any URLs that they receive.
-    if (process.env.NODE_ENV === 'test' || process.env.NODE_ENV === 'development') {
-      alert(`An SMS will be sent, but the link in it will be to production, not to ${window.location.origin}`)
+    if (
+      process.env.NODE_ENV === 'test' ||
+      process.env.NODE_ENV === 'development'
+    ) {
+      alert(
+        `An SMS will be sent, but the link in it will be to production, not to ${window.location.origin}`
+      )
     }
     // On staging, inform devs that sms will not be sent and link must be copy-pasted
     if (process.env.NODE_ENV === 'staging') {
-      alert(`No SMS will be sent, please copy this link ${window.location.origin}`)
+      alert(
+        `No SMS will be sent, please copy this link ${window.location.origin}`
+      )
     }
 
     const { language, sms, token, urls } = this.props
     const url = urls.telephony_url
     const options = {
-      payload: JSON.stringify({to: sms.number, id: this.linkId, language}),
+      payload: JSON.stringify({ to: sms.number, id: this.linkId, language }),
       endpoint: `${url}/v1/cross_device_sms`,
       contentType: 'application/json',
       token: `Bearer ${token}`
     }
-    performHttpReq(options, this.handleResponse , this.handleSMSError)
+    performHttpReq(options, this.handleResponse, this.handleSMSError)
   }
 
-  mobileUrl = ({hosted_sdk_url}) =>
-    // This lets us test the cross device flow locally and on surge.
+  getMobileUrl = () => {
+    const { hosted_sdk_url } = this.props.urls
+    // This lets us test the cross device flow locally and on Surge.
     // We use the same location to test the same bundle as the desktop flow.
-    process.env.MOBILE_URL === "/" ?
+    return process.env.MOBILE_URL === "/" ?
       `${window.location.origin}?link_id=${this.linkId}` :
       `${hosted_sdk_url}/${this.linkId}`
+  }
 
   clearSendLinkClickTimeout() {
     if (this.sendLinkClickTimeoutId) {
       clearTimeout(this.sendLinkClickTimeoutId)
     }
   }
 
+  renderSmsLinkSection = () => {
```

Comment:
Same here. This functional component is pretty big. This file is becoming pretty large in size, I would consider putting this component in a different file

Reply:
As mentioned, send SMS needs a bit more time and care for the refactor so leaving that as it is for now.